### PR TITLE
API

### PR DESCRIPTION
**Pull request by DarkxDev**
- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post.
- Fixed linter errors.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +15,10 @@ class User < ApplicationRecord
   def last_three_posts
     Post.where(author_id: id).last(3)
   end
+
+  private
+
+  def set_user_token
+    self.token = SecureRandom.hex(20)
+  end
 end
```
Comment (BenMukebo) [inline app/models/user.rb:24]: - [OPTIONAL] Great job @dfsalam, after our call I can see the way you guys are generating the token and get access to it 😄 👇🏻 

image

However, there are good ways to do better than your approach. here are some links that could help you in the feature especially when you will be working on a rails API base:
- [Using Rails for API-only Applications](https://guides.rubyonrails.org/api_app.html)
- [Controller Namespaces and Routing](https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing)
- [Rails API + JWT Authentication](https://medium.com/binar-academy/rails-api-jwt-authentication-a04503ea3248)

File: config/routes.rb [hunk 2]
```diff
@@ -19,4 +17,9 @@
 
   delete '/users/:id/posts/:post_id/destroy', to: 'posts#destroy', as: 'delete_post'
   delete '/users/:id/posts/:post_id/:comment_id', to: 'comments#destroy', as: 'delete_comment'
+
+  # API Endpoints
+  get '/users/:id/posts.json', to: 'posts#index'
+  get '/users/:id/posts/:post_id/comments.json', to: 'comments#render_json'
+  post '/api/posts/:post_id/new_comment.json', to: 'api/comments#create'
```
Comment (BenMukebo) [inline config/routes.rb:24]: - [OPTIONAL] You don't need to add .json at the end of the end point.

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @DarkxDev, @dfsalam.** 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- Gitflow was correctly used , keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (DarkxDev) [reply app/models/user.rb:24]: @BenMukebo thanks for the feedback and links to those helpful resources!

**Pull request merged**